generic_config_updater/patch_sorter.py

```diff
@@ -372,40 +372,46 @@ def validate(self, move, diff):
 
 class CreateOnlyMoveValidator:
     """
-    A class to validate create-only fields are only added/removed but never replaced.
-    Parents of create-only fields are also only added/removed but never replaced when they contain
-    a modified create-only field.
+    A class to validate create-only fields are only created, but never modified/updated. In other words:
+    - Field cannot be replaced.
+    - Field cannot be added, only if the parent is added.
+    - Field cannot be deleted, only if the parent is deleted.
     """
     def __init__(self, path_addressing):
         self.path_addressing = path_addressing
 
     def validate(self, move, diff):
-        if move.op_type != OperationType.REPLACE:
-            return True
-
-        # The 'create-only' field needs to be common between current and simulated anyway but different.
-        # This means it is enough to just get the paths from current_config, paths that are not common can be ignored.
-        paths = self._get_create_only_paths(diff.current_config)
         simulated_config = move.apply(diff.current_config)
+        paths = set(list(self._get_create_only_paths(diff.current_config)) + list(self._get_create_only_paths(simulated_config)))
 
         for path in paths:
             tokens = self.path_addressing.get_path_tokens(path)
             if self._value_exist_but_different(tokens, diff.current_config, simulated_config):
                 return False
+            if self._value_added_but_parent_exist(tokens, diff.current_config, simulated_config):
+                return False
+            if self._value_removed_but_parent_remain(tokens, diff.current_config, simulated_config):
+                return False
 
         return True
 
     # TODO: create-only fields are hard-coded for now, it should be moved to YANG models
     def _get_create_only_paths(self, config):
-        if "PORT" not in config:
-            return
+        if "PORT" in config:
+            ports = config["PORT"]
 
-        ports = config["PORT"]
+            for port in ports:
+                attrs = ports[port]
+                if "lanes" in attrs:
+                    yield f"/PORT/{port}/lanes"
 
-        for port in ports:
-            attrs = ports[port]
-            if "lanes" in attrs:
-                yield f"/PORT/{port}/lanes"
+        if "LOOPBACK_INTERFACE" in config:
```

Comment:
>LOOPBACK_INTERFACE [](http://example.com/codeflow?start=12&length=18)

Today we have 2 hard-coded paths, and it may increase in future. Could you make it a general logic on a string array? So in future we can just modify the array items.

Reply:
My thought was to move the `create-only` flag to YANG models directly.

I think I can simplify the existing hard-coding logic, making it more flexible, until we have the soln in YANG.

Reply:
updated